### PR TITLE
CDB-8052, CDB-8050: Upgraded to ruby 3.1.2 (arguments fixes)

### PR DESCRIPTION
**Pull request by ahmad-dev23**
**Motivation**
Wrong number of params in cluster.rb
Fixes for #CPD-8052

**Proposed changes**

**Alternatives considered**

**Testing steps**

### CODE REVIEW

**Review by stephenthedev — APPROVED** [review]

File: aws-flow/lib/aws/decider/generic_client.rb [hunk 1]
```diff
@@ -85,7 +85,7 @@ def bail_if_external
       #      foo.send :bar # plus args and block if appropriate
       #    end
       #
-      def send_async(task, *args, &block)
+      def send_async(task, **args, &block)
```
Comment (kaushik) [inline aws-flow/lib/aws/decider/generic_client.rb]: This looks like it is taking away the functionality for positional arguments for keyword args.

Comment (kaushik) [reply aws-flow/lib/aws/decider/generic_client.rb]: If one splat is used it splits the positional arguments from an array into individual arguments.
But in ruby 3 keyword arguments are treated separately from positional arguments. so *args would not contain keyword arguments that are passed to it. You change allows it to accept keyword arguments but not positional arguments.

Comment (kaushik) [reply aws-flow/lib/aws/decider/generic_client.rb]: I played around with it in pry
```
def async(task, *args, **kwargs)
  send(task, *args, **kwargs)
end
[29] pry(main)> def met3(a, b:)
  p a+b
=> :met3
[33] pry(main)> async(:met3, 2, b: 3)
5
=> 5
```
As you can see I am able to accept and pass both positional and keyword arguments. This worked on both ruby 2.7 and 3.1.2

**Review by kaushik — CHANGES_REQUESTED** [review]
The above suggestion might mean that some of the activity changes might not be needed in the worker PR

**Review by kaushik — APPROVED** [review]

File: aws-flow/Gemfile.lock [hunk 2]
```diff
@@ -0,0 +1,88 @@
+GIT
```
Comment (kaushik) [inline aws-flow/Gemfile.lock]: I don't believe the lock file is part of this repo. Is there a reason you added it now?

**Review by kaushik — CHANGES_REQUESTED** [review]

**Review by kaushik — APPROVED** [review]

**Pull request merged**